src/main/java/frc/robot/subsystems/intakeLimelight/IntakeLimelight.java

```diff
@@ -0,0 +1,579 @@
+  }
+
+  public class NoteDetection {
+    public double latencySec;
+
+    /** Distance from camera */
+    public double distanceMeters;
+
+    /** CCW Yaw Angle */
+    public double yawAngleDeg;
+
+    public NoteDetection(double latencySec, double distanceMeters, double yawAngleDeg) {
+      this.latencySec = latencySec;
+      this.distanceMeters = distanceMeters;
+      this.yawAngleDeg = yawAngleDeg;
+    }
+  }
+
+  /**
+   * Looks for a note
+   *
+   * @return If empty, no note detected. First value is note yaw in degrees (ccw), second value is
+   *     distance in meters.
+   */
+  public Optional<NoteDetection> getNotePos() {
+    // TODO filter low confidence detection in LL dashboard, hopefully
+    if (getPipeline() != pipeline.NOTE_PIPELINE) {
+      setPipeline(pipeline.NOTE_PIPELINE);
+    }
+
+    LimelightHelpers.LimelightResults llresults =
+        LimelightHelpers.getLatestResults(IntakeLimelightConstants.INTAKE_LIMELIGHT_NAME);
+    LimelightTarget_Detector[] targets_Detector = llresults.targetingResults.targets_Detector;
+
+    if (targets_Detector.length == 0) {
+      System.out.println("Didn't see note");
+      return Optional.empty();
+    }
+
+    LimelightTarget_Detector lowestDetection = targets_Detector[0];
+    for (LimelightTarget_Detector detection : targets_Detector) {
+      if (detection.className == "note") {
+        if (detection.ty_pixels > lowestDetection.ty_pixels) { // lower in image = greater y value
+          lowestDetection = detection;
+        }
+      }
+    }
+
+    double angleLimelightToNote =
+        IntakeLimelightConstants.INTAKE_LIMELIGHT_PITCH_DEGREES + lowestDetection.ty;
+    double noteDistanceMeters =
+        (IntakeLimelightConstants.INTAKE_LIMELIGHT_HEIGHT_METERS
+                - Units.inchesToMeters(Constants.NOTE_HEIGHT_INCHES / 2))
+            / Math.tan(Units.degreesToRadians(angleLimelightToNote));
+
+    double yawAngleXDegrees = lowestDetection.tx;
+    double adjustedYawAngleDegrees = yawAngleXDegrees + IntakeLimelightCal.LIMELIGHT_YAW;
+    return Optional.of(
+        new NoteDetection(getLatency(), noteDistanceMeters, adjustedYawAngleDegrees));
+  }
+
+  @Override
+  public void initSendable(SendableBuilder builder) {
+    builder.addDoubleProperty("Target Area", () -> getTargetArea(), null);
+    builder.addDoubleProperty("Skew", () -> getSkew(), null);
+    builder.addDoubleProperty("Latency", () -> getLatency(), null);
+    builder.addDoubleProperty("Distance", () -> getDistanceFromTargetMeters(), null);
+    builder.addDoubleProperty("Tx", () -> getOffSetX(), null);
+    builder.addDoubleProperty("Ty", () -> getOffSetY(), null);
+    builder.addDoubleProperty("Translation Distance", () -> m_lastDistance, null);
+    builder.addDoubleProperty("Translation X", () -> m_lastX, null);
+    builder.addDoubleProperty("Translation Y", () -> m_lastY, null);
+    builder.addBooleanProperty("Valid Target", () -> isValidTarget(), null);
```

Comment:
We might want to remove most of this? Or anything we're not intending on using initially

Reply:
I removed some stuff we definitely aren't using.  Left some stuff that might be useful.